### PR TITLE
Decoupling DB interface from katib manager

### PR DESCRIPTION
**Pull request by achalshant**
Goal: To decouple the DB interface from katib manager in order to make the backend store pluggable. Please refer to the proposal doc for more details - https://docs.google.com/document/d/1xAi4V9SCAVZvrwiVc6a3VpK7BXPx-liQbr6YPdqBThs/edit#heading=h.1hri22e2jta

Major changes:

pkg/db/interface.go was moved to run as a service in cmd/dbif/mysql/main.go
cmd/manager/main.go is now a thin client that connects to the DBIF service using GRPC and operates on the DB.
Some services and messages from api.proto have moved to dbif.proto.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/498)

### CODE REVIEW

File: pkg/api/v1alpha2/build.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 set -x
 set -e
-for proto in api.proto; do
+for proto in dbif/dbif.proto api.proto health/health.proto; do
```
Comment (hougangliu) [inline pkg/api/v1alpha2/build.sh]: `health/health.proto` had been moved out

File: cmd/manager/v1alpha2/main.go [hunk 2]
```diff
@@ -8,123 +8,103 @@ import (
...
 
+/**
+To plug in a new data store, implement and run the service and change the
+dbIfaddress to point to the new service.
+**/
+
 const (
-	port = "0.0.0.0:6789"
+	port        = "0.0.0.0:6789"
+	dbIfaddress = "dbif-mysql:6789"
```
Comment (hougangliu) [inline cmd/manager/v1alpha2/main.go]: please make `dbif` more readable, maybe `mysql-db-backend`?

**Comment by YujiOshima** [comment]
please move `pkg/db/v1alpha2/db_init.go` and `pkg/db/v1alpha2/interface_test.go` to `cmd/dbif/mysql/` since they are only for mysql.

**Comment by achalshant** [comment]
@hougangliu @YujiOshima

**Comment by hougangliu** [comment]
@achalshant thanks for your PR. I am afraid that we cannot include this feature in 0.6 release.
@johnugeorge @richardsliu @gaocegege any comment?

**Comment by richardsliu** [comment]
Agree, we can try to take it up in the next release in June or July.

**Comment by hougangliu** [comment]
@achalshant will you pick up this PR now?

**Comment by achalshant** [comment]
Okay @hougangliu .

**Comment by johnugeorge** [comment]
DB interface is introduced in https://github.com/kubeflow/katib/issues/685

**Comment by johnugeorge** [comment]
Closing this issue as the backend is pluggable

**Pull request closed without merging**